script/cms/compare-json.js

```diff
@@ -502,23 +506,63 @@ const runComparison = () => {
             );
             ++nodesWithDiffs;
           }
+
           if (diff.arrayDiffs?.length > 0) {
             diff.arrayDiffs.unshift(entityFileObject);
             fs.writeFileSync(
               `${defaultFileName}-array.json`,
               JSON.stringify(diff.arrayDiffs.slice(0, MAX_DIFFS), null, 2),
             );
           }
+
+          if (outputHtml) {
+            // Keep track of the subset of pages to validate by HTML comparison.
+            const htmlPagePath = path.join(entity.entityUrl.path, 'index.html');
+            pagesWithDiffs[htmlPagePath] = {
+              ...(diff.deepDiffs?.length && {
+                deepDiff: `${defaultFileName}.json`,
+              }),
+              ...(diff.arrayDiffs?.length && {
+                arrayDiff: `${defaultFileName}-array.json`,
+              }),
+            };
+          }
         }
         ++totalObjectsCompared;
       }
     });
 
+    if (outputHtml) {
```

Comment:
This just crossed my mind, but when the `html` flag is used, do you think we should log the location of `pages-with-diffs.json` to the console? This would make it easier to access and know the location of the file.

Reply:
That's a good idea. Just added it.